src/internal/nativeInterface.ts

```diff
@@ -1,7 +1,14 @@
-import { Platform, NativeModules } from 'react-native';
+import { Platform } from 'react-native';
+import RNDeviceInfoIOS from '../NativeRNDeviceInfoAndroid';
+import RNDeviceInfoAndroid from '../NativeRNDeviceInfoAndroid';
 import { DeviceInfoNativeModule } from './privateTypes';
 
-let RNDeviceInfo: DeviceInfoNativeModule | undefined = NativeModules.RNDeviceInfo;
+let RNDeviceInfo;
+if (Platform.OS === 'ios') {
+  RNDeviceInfo = RNDeviceInfoIOS;
+} else if (Platform.OS === 'android') {
+  RNDeviceInfo = RNDeviceInfoAndroid;
+}
```

Comment:
I think we could also update the `Error` below since it refers to some `react-native <= 0.59` versions that we for sure don't support anymore.  Also, this: `NativeModule.RNDeviceInfo is null.` seems misleading right now.

Reply:
Ok, I'll update it. The goal of the PR was to support the new arch, I didn't look at updating things like this